sql/src/main/java/io/crate/metadata/settings/Validators.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.settings;
+
+import org.elasticsearch.common.Booleans;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.set.Sets;
+
+import java.util.Map;
+import java.util.Set;
+
+public class Validators {
+
+    private static final String UNSUPPORTED_MESSAGE = "Unsupported setting value: ";
+    private static final String INVALID_MESSAGE = "Invalid value for argument '";
+
+    public static Setting.Validator<String> stringValidator(String key, String... allowedValues) {
+        if (allowedValues.length == 0) {
+            return new StringValidator(key);
+        }
+        return new StringValidatorAllowedValuesOnly(key, Sets.newHashSet(allowedValues));
+    }
+
+    private static class StringValidator implements Setting.Validator<String> {
```

Comment:
Maybe rename or comment this to make it clearer which aspect it validates? E.g. what are considered valid strings and what are invalid strings?